### ISSUE TITLE
Websocket forcibly closed by remote host

### ISSUE THREAD

**Issue by simaster123** [danpaquin/coinbasepro-python#363]
Hi All,

Is anyone else receiving the following when using the websockets / have any recommended solutions?

[WinError 10054] An existing connection was forcibly closed by the remote host - data: None

My websocket connection seems to perform fine for about 20 seconds, and then this error occurs.  

Thanks in advance!

**Comment by simaster123** [comment]
Update - It looks like installing cbpro through pip does not provide the most up-to-date websocket_client.py.  I copied and pasted the code available here in GitHub into package websocket_client.py file from pip; that appears to have resolved the issue.

**Comment by simaster123** [comment]
Further update - while the current code available on GitHub stops the issue from occurring every couple minutes, I'm still encountering the issue every 30 minutes or so.  I've added a auto-reconnect workaround when the error is encountered, but it's obviously not ideal.  Anyone know what is causing the error in the first place?

**Comment by edmondja** [comment]
Yes its an issue we encountered. Reconnecting didn't help because sometimes it disconnected very quickly (within seconds) without even telling you right ? Because this is my case. Did you try from scratch with the native coinbase WS api to get the data ?

**Comment by simaster123** [comment]
Same issue re: disconnecting without telling you, although most of my disconnects are of the type in my prompt - forcibly closed.  I added some additional checks in my code to confirm whether messages are being received.  It can be a couple minutes before I'm able to re-establish a somewhat stable connection.  I haven't tried from scratch yet.  Any ideas as to what's causing the issues?

**Comment by noah222** [comment]
See attached script and run it as a place to start. It's not my final code but a stable thing that works ALL THE TIME. Note that it requires you to create and point to some text files for logging. 
[Matches_feed_demo.txt](https://github.com/danpaquin/coinbasepro-python/files/3140429/Matches_feed_demo.txt)

**Comment by noah222** [comment]
I thought I would describe also my method for working with the above code. I log all prices every second. This is running in it's own window, even if I am not trading. I then use this granular data for backtesting with my own trading script. I can hit play and run the market moves over and over until I understand them.

So this above script is also updating a text file with the current price. I had an Arduino project that would load this price as a fun LED display on my desk, but I gave that to someone as a gift.

So what the trading script does is it just looks at the textfile to get the current price (instead of connecting to the feed directly). Then with the feed goes down or gets wonky, my complex trading state is preserved and does not crash out. 

But if you use the above method as a starting place, you can have an unbreakable system that cannot be stopped, even if your internet blips out for a sec.

**Comment by mcardillo55** [comment]
Are you still seeing this issue @simaster123 ?

**Comment by sundarkast** [comment]
Seeing a similar issue - created a new one before I noticed this

**Comment by sundarkast** [comment]
> websocket_client.py

Which one is being used?

 ls -al /usr/local/lib/python3.6/dist-packages/cbpro/websocket_client.py
4812 Dec 12 23:01 /usr/local/lib/python3.6/dist-packages/cbpro/websocket_client.py
 ls -al /usr/local/lib/python3.6/dist-packages/gdax/websocket_client.py
3187 Dec 12 23:01 /usr/local/lib/python3.6/dist-packages/gdax/websocket_client.py

**Comment by pbpunk** [comment]
I am also wrestling with this error. It didn't seem to be a problem when I was subscribed to the ticker channel, but every since I started using level 2 it has become a problem.

**Comment by jacobfrank250** [comment]
Is this the same as #341? If so, I too am experiencing the issue.

**Comment by noah222** [comment]
Also, there is no reason to use the cbpro websocket client. To me it seems clunky. Make your own connections using websocket for better control of the process. Its only a few lines of code to subscribe and resubscribe at will. Then find a way (hint, memcashe) to keep the feed running in its own window for monitoring so it can be restarted when problems arise. Thats all I can say about the feed. It seems silly to be connecting all the time, but thats how you ensure it works with these limited tools.

**Comment by gabdim** [comment]
How can I disconnect the websocket and reconnect from time to time using this python wrapper?
